### PR TITLE
TMDM-14650 [CVE] - Update Jackson to 2.10.1 (refactor)

### PR DESCRIPTION
**Pull request by lpw82gf**
https://jira.talendforge.org/browse/TMDM-14650
All artificates with groupId start with `com.fasterxml.jackson` have same parent pom as bellow, so define one parameter `jackson.version` would be enough.
```
<parent>
    <groupId>com.fasterxml.jackson</groupId>
    <artifactId>jackson-base</artifactId>
    <version>2.10.1</version>
  </parent>
```

### CODE REVIEW

File: org.talend.mdm.core/pom.xml [hunk 1]
```diff
@@ -257,32 +271,17 @@
...
-		</dependency>
+            <version>${jackson.version}</version>
+        </dependency>
         <dependency>
-            <groupId>com.fasterxml.jackson.jaxrs</groupId>
-            <artifactId>jackson-jaxrs-json-provider</artifactId>
-            <version>${jackson.provider.version}</version>
+            <groupId>com.fasterxml.woodstox</groupId>
+            <artifactId>woodstox-core</artifactId>
+            <version>${woodstox.core.version}</version>
```
Comment (hw-zhu) [inline org.talend.mdm.core/pom.xml:279]: woodstox-core depend on stax2-api-4.2.jar, but we have the third dependence in https://github.com/Talend/tmdm-server-se/blob/master/org.talend.mdm.core/pom.xml#L451, the scope is runtime,  actually, the jar exists under  talendmdm/lib.

File: org.talend.mdm.core/pom.xml [hunk 2]
```diff
@@ -232,22 +231,37 @@
...
 	            </exclusion>
             </exclusions>
-            <version>${jackson.provider.version}</version>
+            <version>${jackson.version}</version>
         </dependency>
         <dependency>
             <groupId>com.fasterxml.jackson.dataformat</groupId>
             <artifactId>jackson-dataformat-xml</artifactId>
-            <version>${jackson.provider.version}</version>
+            <version>${jackson.version}</version>
```
Comment (hw-zhu) [inline org.talend.mdm.core/pom.xml:239]: As the jar, version 2.10.1 depend on woodstox-core [6.0.2 ~ 6.2.1], but its current version is 5.3.0, if need to upgrade it together?

File: org.talend.mdm.core/pom.xml [hunk 3]
```diff
@@ -232,22 +231,37 @@
...
         </dependency>
         <dependency>
             <groupId>com.fasterxml.jackson.datatype</groupId>
             <artifactId>jackson-datatype-joda</artifactId>
-            <version>${jackson.provider.version}</version>
+            <version>${jackson.version}</version>
```
Comment (hw-zhu) [inline org.talend.mdm.core/pom.xml:244]: jackson-datatype-jode depend on joda-time, so I suggest exclusion the lower version joda-time.
```
            <exclusions>
                <exclusion>
                    <groupId>joda-time</groupId>
                    <artifactId>joda-time</artifactId>
                </exclusion>
            </exclusions>
```
using directly below 2.10.

File: org.talend.mdm.core/pom.xml (hunk 2 above)
Comment (lpw82gf) [reply org.talend.mdm.core/pom.xml:239]: Let me upgrade and run the test again.

File: org.talend.mdm.core/pom.xml (hunk 3 above)
Comment (lpw82gf) [reply org.talend.mdm.core/pom.xml:244]: No need to exclude it explictly, since in the WAR 2.10 will be used already, and 2.9.9 has not CVE issue too.

Comment (lpw82gf) [reply org.talend.mdm.core/pom.xml:244]: But exclude it is also OK.

File: org.talend.mdm.core/pom.xml (hunk 1 above)
Comment (lpw82gf) [reply org.talend.mdm.core/pom.xml:285]: If the class in a jar not imported directly in mdm's code, we can all add `<scope>runtime</scope>`, stax2-api is just one of them, it is more strict defination. No need to remove.

File: org.talend.mdm.core/pom.xml (hunk 2 above)
Comment (lpw82gf) [reply org.talend.mdm.core/pom.xml:239]: Upgrade and tested OK.

**Pull request merged**